### PR TITLE
Core need some love.

### PR DESCRIPTION
**Pull request by FrankFFFF**
Keep reading and get familiar with it. :neckbeard:

### CODE REVIEW

**Comment by ADmad** [comment]
Seems you went a bit overboard with your love. Your changes have caused failing tests.

**Comment by Fieah** [comment]
LOL, will check.

File: src/Core/Configure/FileConfigTrait.php [hunk 1]
```diff
@@ -46,14 +46,7 @@ protected function _getFilePath($key, $checkExists = false)
...
         list($plugin, $key) = pluginSplit($key);
-
-        if ($plugin) {
-            $file = Plugin::configPath($plugin) . $key;
-        } else {
-            $file = $this->_path . $key;
-        }
-
-        $file .= $this->_extension;
+        $file = ($plugin ? Plugin::configPath($plugin) : $this->_path) . $key . $this->_extension;
```
Comment (ADmad) [inline src/Core/Configure/FileConfigTrait.php]: Using ternary and concatenation in single statement is not good for readability. Keep the concatenation separate.

File: src/Core/functions.php [hunk 2]
```diff
@@ -153,13 +153,9 @@ function pr($var)
      */
     function pj($var)
     {
-        if (!Configure::read('debug')) {
```
Comment (ADmad) [inline src/Core/functions.php]: This should not be removed. Throughout the core we try and return early from method when possible.

Comment (Fieah) [inline src/Core/functions.php]: Actually this is same with `pr()` , so i chose to follow it.

File: src/Core/Configure/FileConfigTrait.php (hunk 1 above)
Comment (Fieah) [inline src/Core/Configure/FileConfigTrait.php]: OK, will revert this.

File: src/Core/functions.php (hunk 2 above)
Comment (ADmad) [inline src/Core/functions.php]: Then it would be better to change `pr()` instead.

Comment (Fieah) [inline src/Core/functions.php]: OK.
I'm not sure what caused this to break, should i make another pull request or just change in here?

File: src/Core/StaticConfigTrait.php [hunk 3]
```diff
@@ -72,15 +72,18 @@
...
-        // Read config.
-        if ($config === null && is_string($key)) {
-            return isset(static::$_config[$key]) ? static::$_config[$key] : null;
-        }
-        if ($config === null && is_array($key)) {
-            foreach ($key as $name => $settings) {
-                static::config($name, $settings);
+        if ($config === null) {
+			// Read config.
+			if (is_string($key)) {
```
Comment (antograssiot) [inline src/Core/StaticConfigTrait.php]: indentations seems of (should use space)

File: src/Core/functions.php (hunk 2 above)
Comment (ADmad) [inline src/Core/functions.php]: You can make change to this PR itself and at the end squash commits.

File: src/Core/StaticConfigTrait.php (hunk 3 above)
Comment (Fieah) [inline src/Core/StaticConfigTrait.php]: Hi, had noticed this just now, thanks :)

**Comment by Fieah** [comment]
`FILE: /home/travis/build/cakephp/cakephp/src/Core/functions.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 141 | ERROR | Expected 0 blank lines before closing function brace;`
This gotta be kidding me, since when this became an issue?

**Comment by Fieah** [comment]
`zend_mm_heap corrupted`
This is not related, right? Is there another way to make it re-run?
Thanks.

**Comment by AD7six** [comment]
@Fieah what's "it"? Depends whether it's persistent as to whether it's _probably_ related or not.

Any commit/change will start a new build, team members can them (as I just did for the sqlite run).

**Comment by Fieah** [comment]
@AD7six  Hi,
Yeah, i mean re-run Travis, now everything seems to be working fine, thank you :)

**Comment by rchavik** [comment]
Umm,  it's good that love is all around. But what is this change trying to improve?

File: src/Core/functions.php [hunk 4]
```diff
@@ -156,11 +158,9 @@ function pj($var)
         if (!Configure::read('debug')) {
             return;
         }
-        if (PHP_SAPI === 'cli') {
-            printf("\n%s\n\n", trim(json_encode($var, JSON_PRETTY_PRINT)));
-        } elseif (Configure::read('debug')) {
-            printf('<pre class="pj">%s</pre>', trim(json_encode($var, JSON_PRETTY_PRINT)));
-        }
+
+        $template = PHP_SAPI !== 'cli' ? '<pre class="pr">%s</pre>' : "\n%s\n\n";
```
Comment (AD7six) [inline src/Core/functions.php]: the pre class name has changed here

**Comment by rchavik** [comment]
If it's not clear, I'm not opposed to the changes.  But I hope you'll amend the commit message to explain what a particular commit does.  The current ones are amusing easter eggs, but ... :)

**Comment by AD7six** [comment]
@Fieah when you're happy with the result, can you squash your changes into one commit giving it a descriptive commit title and description please.

:+1: otherwise.

File: src/Core/functions.php (hunk 4 above)
Comment (Fieah) [inline src/Core/functions.php]: Nice catch, another boo boo made by me. 
Thank you ;)

**Comment by Fieah** [comment]
@rchavik  @AD7six 
Yeah, will make another PR later, this one is messy.
Thanks for the help :)

**Comment by AD7six** [comment]
@Fieah you don't need to create a new pr.

    git fetch
    git rebase origin/master # read 
    git push -f

and reopen this PR.

**Comment by Fieah** [comment]
@AD7six 
Thank you, will try. :+1:

**Comment by thaJeztah** [comment]
@Fieah instructions above are correct, but you need to **first** reopen the PR, **then** `git push -f`, because GitHub will not update a PR if it's closed

**Comment by Fieah** [comment]
Hi @thaJeztah 
OK, trying now, thank you. :)

**Comment by markstory** [comment]
Changes all seem reasonable. I personally find less deeply nested control structures easier to work with, but it does result in some duplicated conditions.

File: src/Core/StaticConfigTrait.php [hunk 5]
```diff
@@ -72,15 +72,18 @@
...
+            if (is_array($key)) {
+                foreach ($key as $name => $settings) {
+                    static::config($name, $settings);
+                }
+                return;
             }
-            return;
         }
 
         if (isset(static::$_config[$key])) {
```
Comment (thaJeztah) [inline src/Core/StaticConfigTrait.php]: Not related to this PR, but this will break if `$key` is an array?

**Comment by Fieah** [comment]
FINALLY, commits DONE, ci PASS.
Thanks a lot guys, i have learned something new today. :raised_hands:

File: src/Core/StaticConfigTrait.php (hunk 5 above)
Comment (markstory) [inline src/Core/StaticConfigTrait.php]: Yes if config is not null and key is an array bad things will happen.

File: src/Core/Configure/FileConfigTrait.php (hunk 1 above)
Comment (Fieah) [inline src/Core/Configure/FileConfigTrait.php]: Hmm...6 lines to 1 lines with spaces.
I'm still wondering how bad this can be. :8ball:
